File: src/main/java/com/bz/springkafka/springkafka/controller/KafkaMessageController.java
package com.bz.springkafka.springkafka.controller;

import com.bz.springkafka.springkafka.kafka.KafkaProducerService;
import com.bz.springkafka.springkafka.storage.AppStorage;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/messages")
@RequiredArgsConstructor
public class KafkaMessageController {

    private final KafkaProducerService kafkaProducerService;

    @PostMapping
    public String sendMessage(@RequestBody String message) {
        return kafkaProducerService.sendMessage("my-topic", message);
    }

    @GetMapping("/{id}")
    public String readMessage(@PathVariable String id) {
        return AppStorage.get(id);
    }

}

File: src/main/kotlin/com/bz/springkafka/controller/KafkaMessageController.kt
package com.bz.springkafka.controller

import com.bz.springkafka.kafka.KafkaProducerService
import com.bz.springkafka.storage.AppStorage
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/messages")
class KafkaMessageController(private val kafkaProducerService: KafkaProducerService) {

    @PostMapping
    fun sendMessage(@RequestBody message: String): String {
        return kafkaProducerService.sendMessage("my-topic", message)
    }

    @GetMapping("/{id}")
    fun readMessage(@PathVariable id: String): ResponseEntity<String> {
        return AppStorage.get(id)?.let { ResponseEntity.ok(it) } ?: ResponseEntity.notFound().build()
    }
}
